src/klein/test/_strategies.py

```diff
@@ -61,3 +116,125 @@ def latin1_text(draw, min_size=None, max_size=None):
         characters(max_codepoint=255),
         min_size=min_size, max_size=max_size,
     )))
+
+
+@composite
+def idna_text(draw, min_size=None, max_size=None):  # pragma: no cover
+    # type: (DrawCallable, Optional[int], Optional[int]) -> Text
+    """
+    A strategy which generates IDNA-encodable text.
+
+    @param min_size: The minimum number of characters in the text.
+        C{None} is treated as C{0}.
+
+    @param max_size: The maximum number of characters in the text.
+        Use C{None} for an unbounded size.
+    """
+    return draw(text(
+        min_size=min_size, max_size=max_size, alphabet=idna_characters()
+    ))
+
+
+@composite
+def port_numbers(draw, allow_zero=False):  # pragma: no cover
+    # type: (DrawCallable, bool) -> int
+    """
+    A strategy which generates port numbers.
+
+    @param allow_zero: Whether to allow port C{0} as a possible value.
+    """
+    if allow_zero:
+        min_value = 0
+    else:
+        min_value = 1
+
+    return draw(integers(min_value=min_value, max_value=65535))
+
+
+@composite
+def hostname_labels(draw, allow_idn=True):  # pragma: no cover
+    # type: (DrawCallable, bool) -> Text
+    """
+    A strategy which generates host name labels.
+
+    @param allow_idn: Whether to allow non-ASCII characters as allowed by
+        internationalized domain names (IDNs).
+    """
+    if allow_idn:
+        label = draw(idna_text(min_size=1, max_size=63))
+    else:
+        label = draw(text(
+            min_size=1, max_size=63,
+            alphabet=unicode(ascii_letters + digits + u"-")
+        ))
+
+    return label
+
+
+@composite
+def hostnames(
+    draw, allow_leading_digit=True, allow_idn=True
+):  # pragma: no cover
+    # type: (DrawCallable, bool, bool) -> Text
+    """
+    A strategy which generates host names.
+
+    @param allow_leading_digit: Whether to allow a leading digit in host names;
+        they were not allowed prior to RFC 1123.
+
+    @param allow_idn: Whether to allow non-ASCII characters as allowed by
+        internationalized domain names (IDNs).
+    """
+    labels = draw(
+        lists(
+            hostname_labels(allow_idn=allow_idn),
+            min_size=1, max_size=5, average_size=2
+        )
+        .filter(lambda ls: sum(len(l) for l in ls) + len(ls) - 1 <= 252)
+    )
+
+    name = u".".join(labels)
+
+    return name
+
+
+_path_characters = tuple(
+    unichr(i)
+    for i in range(maxunicode) if i not in (ord(c) for c in "#/?")
+)
+
+@composite
+def path_segments(draw):  # pragma: no cover
+    # type: (DrawCallable) -> Text
+    """
+    A strategy which generates URL path segments.
+    """
+    return draw(
+        iterables(
+            text(min_size=1, alphabet=_path_characters),
+            max_size=10, average_size=3,
+        )
+    )
+
+
+@composite
```

Comment:
This seems like something that would make sense to add to the `hyperlink` package. If this is for expedience, so as not to wait for a hyperlink release, this makes sense.

Reply:
yes, agreed